source/libtomo/misc/remove_ring.c

```diff
@@ -629,7 +629,7 @@ remove_ring(float* data, float center_x, float center_y, int dx, int dy, int dz,
         polar_image =
             polar_transform(image, center_x, center_y, dx, dy, &pol_width, &pol_height,
                             thresh_max, thresh_min, r_scale, ang_scale, ring_width);
-        m_azi = ceil((float) pol_height / 360.0) * angular_min;
+        m_azi = floor((float) pol_height / 360.0 * angular_min);
```

Comment:
Could you explain more about why changing from ceil to floor relates to L273 and where the out of bounds error actually occurs?

```suggestion
        m_azi = ceil((float) pol_height / 360.0 * angular_min);
        // angular_min is float instead of int
```
Would switching the type of `angular_min` to allow `float` inputs also work? This way would be backward compatible.

Reply:
We must switch from ceil to floor because otherwise, the strictly less condition is not met for heights smaller than 179. i.e.
```
ceil((float) height / 360.0 * (2 * 179)) < height
```
is not true for small height.